### PR TITLE
Features/input field

### PR DESCRIPTION
**Pull request by ragg09-sunasterisk**
## Issue Link
[LMS-316](https://framgiaph.backlog.com/view/LMS-316)

## Defintion of Done
- should be able to use input field dynamically

## Steps to reproduce
- Import InputFiled component
- pass props necessary 
- http://localhost:3000/demo/components#inputField

## Affected Components / Functionalities / Page
n/a

## Test Cases
_will update soon..._

## Notes
n/a

## Screenshots
image
image

### CODE REVIEW

**Review by Marielleregner7 — APPROVED** [review]
**Approved**
- able to use input fields dynamically in the provided page http://localhost:3000/demo/components#inputField 
- No ESLint warnings or errors.

Comments: 0
Test time: 10 mins
Iterations: 0

File: client/src/shared/components/InputField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  heightProp?: string;
+}
+
+const InputField: React.FC<InputFieldProps> = ({
+  label,
+  placeholder,
+  type,
+  id,
+  widthProp,
+  heightProp
```
Comment (escbooster12-dev) [inline client/src/shared/components/InputField/index.tsx:19]: - no need to add postfix `prop`.
```suggestion
  width,
  height
```

File: client/src/shared/components/InputField/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import React from 'react';
+
+interface InputFieldProps {
+  label?: string;
+  placeholder?: string;
+  type?: string;
```
Comment (escbooster12-dev) [inline client/src/shared/components/InputField/index.tsx:7]: - set the default value to `text` for `type`

File: client/src/pages/demo/components/index.tsx [hunk 3]
```diff
@@ -97,6 +100,46 @@ const DemoComponent: React.FunctionComponent = () => {
...
+
+            <div className="bg-gray-300 p-[5px]">
+              <p>label: (string)[""] ex. label="Testing"</p>
+              <p> placeholder: (string)[""] ex. placeholder="Testing"</p>
+              <p>
+                type: (string)["text"] ex. type="text | email | passwrod | etc"
+              </p>
+              <p> id: (string)[""] ex. label="Testing"</p>
+              <p> widthProp: "100%" | "50px",</p>
+              <p>heightProp: "100%" | "50px"</p>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:139]: - update prop name

File: client/src/shared/components/InputField/index.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  id?: string;
+  width?: string;
+  height?: string;
+}
+
+const InputField: React.FC<InputFieldProps> = ({
+  label,
+  placeholder,
```
Comment (escbooster12-dev) [inline client/src/shared/components/InputField/index.tsx:15]: - these two props should be optional.

File: client/src/shared/components/InputField/index.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  return (
+    <div className="mb-4">
+      <label
+        className="block text-gray-700 text-sm font-bold mb-2"
+        htmlFor={id}
+      >
+        {label}
+      </label>
```
Comment (escbooster12-dev) [inline client/src/shared/components/InputField/index.tsx:33]: - it should not render a `label` html tag. if no label prop is passed.

File: client/src/shared/components/InputField/index.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable object-shorthand */
+import React from 'react';
+
+interface InputFieldProps {
```
Comment (escbooster12-dev) [inline client/src/shared/components/InputField/index.tsx:4]: - this interface could be usable in the future development. so please export it.
```suggestion
export interface InputFieldProps {
```

**can you kindly apply the same thing as well in the other existing components**

File: client/src/shared/components/InputField/index.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  height
+}: InputFieldProps) => {
+  const propStyle = {
+    width: width,
+    height: height
+  };
+
+  return (
+    <div className="mb-4">
+      {label?.toString() !== '' && (
```
Comment (escbooster12-dev) [inline client/src/shared/components/InputField/index.tsx:28]: - simplify 
```suggestion
      {label && (
```

Comment (ragg09-sunasterisk) [reply client/src/shared/components/InputField/index.tsx:28]: okay sir, Ill just add this ESLint rule `/* eslint-disable @typescript-eslint/strict-boolean-expressions */`

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
✅Able to use input fields dynamically in the provided page http://localhost:3000/demo/components#inputField
✅ No ESLint warnings or errors.

Comments: 10
Test time: 35 mins
Iterations: 2

**Review by kdacudag — APPROVED** [review]
Approved

**Comment by Marielleregner7** [comment]
comments: 12
iterations: 3

**Pull request merged**